sdk/core/azure-core/azure/core/pipeline/transport/pyodide.py

```diff
@@ -0,0 +1,160 @@
+# --------------------------------------------------------------------------
+#
+# Copyright (c) Microsoft Corporation. All rights reserved.
+#
+# The MIT License (MIT)
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the ""Software""), to
+# deal in the Software without restriction, including without limitation the
+# rights to use, copy, modify, merge, publish, distribute, sublicense, and/or
+# sell copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS
+# IN THE SOFTWARE.
+#
+# --------------------------------------------------------------------------
+
+from collections.abc import AsyncIterator
+from io import BytesIO
+
+import js # pylint: disable=import-error
+from pyodide import JsException  # pylint: disable=import-error
+from pyodide.http import pyfetch  # pylint: disable=import-error
+
+from azure.core.exceptions import HttpResponseError
+from azure.core.pipeline import Pipeline
+from azure.core.utils import CaseInsensitiveDict
+
+from ...rest._http_response_impl_async import AsyncHttpResponseImpl
+from . import HttpRequest
+from ._requests_asyncio import AsyncioRequestsTransport
+
+
+class PyodideTransportResponse(AsyncHttpResponseImpl):
+    """Async response object for the `PyodideTransport`."""
+
+    def __init__(self, **kwargs):
+        super(PyodideTransportResponse, self).__init__(**kwargs)
+        # clone to avoid reading from the same `FetchResponse` a second time in `load_body`.
+        self._js_stream = self.internal_response.clone().js_response.body
+        self._js_reader = None
+
+    async def close(self) -> None:
+        """We don't actually have control over closing connections in the browser, so we just pretend
+        to close.
+        """
+        self._is_closed = True
+
+    async def load_body(self) -> None:
```

Comment:
is your response supposed to only be compatible with `azure.core.rest.AsyncHttpResponse`? If so, you don't have to do anything for `load_body` and `body`, since they're not part of the `azure.core.rest.AsyncHttpResponse` protocol. See [here](https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/azure/core/rest/_rest_py3.py#L377) for what the documented api surface is